### ISSUE TITLE
No audio output on 3.5mm audiojack on rpi4 aaos/aosp 14

### ISSUE THREAD

**Issue by ggdover** [raspberry-vanilla/android_local_manifest#147]
I've so far had success with audio output from 3.5mm audio jack on Raspberry pi 4 model B while using the android-13 branch.
However when I upgraded to using android-14 I'm no longer getting any audio output from the 3.5mm audio jack.
I've tested both the car (aosp_rpi4_car) and normal (aosp_rpi4-userdebug) version but neither have any audio on the 3.5mm jack. I am however able to get audio when I plug in a speaker via usb.

With the investigation I've done so far I can see that the audio_hal library for usb is being called
(functions like "adev_open_output_stream" and "out_write" in "hardware/libhardware/modules/usbaudio/audio_hal.c")
but the same functions in "audio_hw.c" in "device/brcm/rpi4/audio" is not called on android 14 like they are on android 13, which is my guess why the audio is audible in usb device but not 3.5mm jack on android 14.

I've also read in aosp documentation about how Audio HAL is moving from using HIDL to use AIDL instead for Audio HAL in android (https://source.android.com/docs/core/audio/aidl-implement) so I wonder if that could have anything to do with it,
but I have no more specific evidence to suggest that.

**Comment by KonstaT** [comment]
I can't reproduce such issue. Most likely you have some USB audio device connected in which case you're not even supposed to get audio output from the 3.5mm jack.

Just to be sure, please provide `logcat` and output of `cat /proc/asound/cards` and `getprop | grep audio`.

**Comment by KonstaT** [comment]
Closing as there has been no update in a month. No additional information was provided.

Assuming resolved by removing any connected USB audio device. Tablet UI builds also have an option to use the built-in audio device (3.5mm jack, HDMI, DAC) even when there's a USB audio device connected (Settings -> Sound & vibration -> Audio will play on -> This tablet).

**Comment by ggdover** [comment]
Sorry for lack of response.
After checking available alsa devices by running `cat /proc/asound/cards`
I noticed that the device for 3.5mm jack did not appear, only the hdmi.
This was due to a fault in the linux kernel, which I build myself since I wanted
to enable some features there.

So it's correct to close this issue.

TLDR my issue was due to fault in my Linux kernel build.
